test/es_test_case.py

```diff
@@ -40,16 +40,6 @@ def setUpClass(cls):
         except BaseException:  # no coverage
             cls._kill_containers()
             raise
-        else:
-            # By default `max_compilations_per_minute` is set to 15, we must increase that threshold in order to execute
-            # dynamic header row and HashCode generation for test_full_hash_validity.
-            # If this also happens in production code, we should move this code to be run as part of `make terraform`,
-            # as a provisioner, for example. Or we can precompile the Painless scripts as part of `make deploy`.
-            cls.es_client.cluster.put_settings(body={
-                'persistent': {
-                    'script.max_compilations_per_minute': 60
```

Comment:
It appears the setting has been renamed, but still has the same default of 15/min (well, 75/5min to be precise, which seems to be equivalent). How is it possible that we can completely remove this override even though the same default rate applies?

Reply:
You could technically say they are equivalent but they are different non the less. The period for circuit breaker is now more ample for invoking the script, compile it and run it without exhausting the allowed rate. The compiling of the script is only done for manifest generation requests, and since that portion of our code does not run constantly we can get by with removing it. Besides, if you recall this was a problem that would manifest it self occasionally when running manifest object key generation tests. Which to me indicates that we might have been right on the limit of the exhaustion, extending the period benefited us.

Reply:
That is true. It is also true that 
```
60/5 + 10/5 + 0/5 + 2/5 + 3/5 == 75/5
```
The concern isn't mathematical but rather temporal. The average of 75/5min over a period of a minute is 15/1min, however we are not constrained to 1 minute but instead we are constrained to 5 minutes. We must compile our scripts less than 75 times before the period of 5 minutes. Taking into account that we compile scripts occasionally when Service is required to obtain a content-hash from Elasticsearch for a given manifest, we know that this process is not continuous e.i. it does not happen more than 75 times in 5 minutes. Unless of course Service is bombarded with more than 75 manifest request within a period of 5 minutes.
So if each of the following fraction represents the number of compilations in one minute over a span of five minute e.i. 5  fractions.
```
60/5, 0/5, 0/5, 10/5, 5/5
```
We can see that it is possible to compile at least 60 times in one minute leaving a pool of 15 compilations for the rest of the 4 minutes. Therefore, not requiring an expansion on the default limit of compilations. That is my reasoning, perhaps there is something that I am missing, 🤔 in which case I would appreciate if we discuss it in a teleconference.

Reply:
Nope, that makes sense.